kms/mackms/mackms.go

```diff
@@ -966,10 +1075,29 @@ func storeCertificate(label string, cert *x509.Certificate) error {
 
 	// Adding the label here doesn't have any effect. Apple Keychain always uses
 	// the commonName.
-	attributes, err := cf.NewDictionary(cf.Dictionary{
+	dict := cf.Dictionary{
 		security.KSecClass:    security.KSecClassCertificate,
 		security.KSecValueRef: certRef,
-	})
+	}
+	// Apple recommends to set KSecUseDataProtectionKeychain parameters always
+	// to true for all keychains operations. This option requires to use a
+	// code-signed application.
+	if u.useDataProtectionKeychain {
+		dict[security.KSecUseDataProtectionKeychain] = cf.True
+		// The data in the keychain item cannot be accessed after a restart
```

Comment:
Is it possible to detect the required entitlements are available in the program? If so, when the intention is to use the data protection keychain, and the application isn't codesigned / missing entitlements, can it return an error indicating that case? This may not be necessary if errors from the OS are clear enough, but I've seen enough OSError codes that weren't overly clear 😅